src/System.Private.CoreLib/shared/System/IO/FileStreamCompletionSource.Win32.cs

```diff
@@ -198,12 +201,12 @@ private void CompleteCallback(ulong packedResult)
                 }
             }
 
-            private static void Cancel(object state)
+            private static void Cancel(object? state)
```

Comment:
Why is state nullable here? If null is passed the Debug.Assert would fail, right?

Reply:
Was it to avoid adding a bunch of `!` on call sites?

Reply:
> Why is state nullable here?

To match the delegate it's wrapped in, which matches the ctor that delegate is passed to.  If/when we get attributes, we can potentially unwind all of that, but for now, this keeps the suppression (the assert) closest to where it would affect things if there were a problem.